src/shared/logger/commands.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Logger } from '.'
+import { recordVscodeViewLogs } from '../telemetry/telemetry'
+import { Commands } from '../vscode/commands2'
+
+function revealLines(editor: vscode.TextEditor, start: number, end: number): void {
+    const startPos = editor.document.lineAt(start).range.start
+    const endPos = editor.document.lineAt(end - 1).range.end
+
+    editor.selection = new vscode.Selection(startPos, endPos)
+    editor.revealRange(new vscode.Range(startPos, endPos))
+}
+
+function clearSelection(editor: vscode.TextEditor): void {
+    const start = new vscode.Position(0, 0)
+
+    editor.selection = new vscode.Selection(start, start)
+}
+
+const from = Commands.instance.from.bind(Commands.instance)
+
+export class Logging {
+    public static readonly declared = {
+        viewLogs: from(this).declareOpenLogUri('aws.viewLogs'),
+        viewLogsAtMessage: from(this).declareOpenLogId('aws.viewLogsAtMessage'),
+    }
```

Comment:
Still thinking of the best way to organize command declarations. Adding them to the class works fine but I don't think every sub-system needs a dedicated class. 

You can also do:
```ts
export const viewLogs: declare('aws.viewLogs', (defaultLogUri: vscode.Uri) => {
    return (logUri = defaultLogUri) => {
        // impl.
    }
})
```

Though obviously if there's a lot of shared context then a class makes more and more sense.